### PR TITLE
Add files via upload

### PR DESCRIPTION
**Pull request by Tanuj2004**

### CODE REVIEW

File: TileCanvas.java [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+        graphics2D.setColor(Color.BLACK);
+        graphics2D.fillRect(0, 0, getWidth(), getHeight() - 50);
+        int tileSize = (getWidth() - 20) / gridSize;
+        graphics2D.setFont(new Font("Arial", Font.BOLD, tileSize / 3));
+        for (int row = 0; row < gridSize; row++) {
+            for (int col = 0; col < gridSize; col++) {
+                drawTile(graphics2D, row, col, tileSize);
+            }
+        }
+    }
```
Comment (kritika-kumari-git) [inline TileCanvas.java:59]: Do not use documentation for the methods whose name describes the method itself.

File: TileCanvas.java [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+        int x = col * tileSize + 10;
+        int y = row * tileSize + 10;
+        if (isAnimating) {
+            double scale = 1.0 + 0.1 * Math.sin(Math.PI * animationStep / 20);
+            int scaledSize = (int) (tileSize * scale);
+            int offset = (tileSize - scaledSize) / 2;
+            x += offset;
+            y += offset;
+            tileSize = scaledSize;
+        }
```
Comment (KRITIKA-KUMARI18) [inline TileCanvas.java:74]: Do not use documentation for defining the methods. Instead use Single Line Comments

File: TileCanvas.java [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+     * Updates the board and the score and triggers an animation.
+     */
+    public void updateBoard(int[][] gameBoard, int currentScore) {
+        this.gameBoard = gameBoard;
+        this.currentScore = currentScore;
+        this.isAnimating = true;
+        this.animationStep = 0;
+        animationTimer.start();
+    }
+}
```
Comment (kritika-kumari-git) [inline TileCanvas.java:137]: Again. Also mention return type and parameters of the method.

File: TileFusionManager.java [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+                    if (newRow < gridSize - 1 && gameBoard[newRow + 1][col] == gameBoard[newRow][col]) {
+                        gameBoard[newRow + 1][col] *= 2;
+                        currentScore += gameBoard[newRow + 1][col];
+                        gameBoard[newRow][col] = 0;
+                        hasMoved = true;
+                    }
+                }
+            }
+        }
+    }
```
Comment (kritika-kumari-git) [inline TileFusionManager.java:161]: Also mention return type and parameters of the method.

File: TileFusionManager.java [hunk 5]
```diff
@@ -0,0 +1,234 @@
...
+                    if (newCol < gridSize - 1 && gameBoard[row][newCol + 1] == gameBoard[row][newCol]) {
+                        gameBoard[row][newCol + 1] *= 2;
+                        currentScore += gameBoard[row][newCol + 1];
+                        gameBoard[row][newCol] = 0;
+                        hasMoved = true;
+                    }
+                }
+            }
+        }
+    }
```
Comment (kritika-kumari-git) [inline TileFusionManager.java:209]: Also mention return type and parameters of the method.

File: TileFusionManager.java [hunk 6]
```diff
@@ -0,0 +1,234 @@
...
+     */
+    private void resetGame() {
+        gameBoard = new int[gridSize][gridSize];
+        currentScore = 0;
+        // Initialize the board with two starting tiles
+        gameBoard[0][0] = 2;
+        gameBoard[1][1] = 2;
+        tileCanvas.updateBoard(gameBoard, currentScore);
+    }
+}
```
Comment (kritika-kumari-git) [inline TileFusionManager.java:234]: Do not use documentation for the methods whose name describes the method itself.

File: TileFusionManager.java [hunk 7]
```diff
@@ -0,0 +1,234 @@
...
+                if (j > 0 && gameBoard[i][j] == gameBoard[i][j - 1]) {
+                    return false;
+                }
+                if (j < gridSize - 1 && gameBoard[i][j] == gameBoard[i][j + 1]) {
+                    return false;
+                }
+            }
+        }
+        return true;
+    }
```
Comment (kritika-kumari-git) [inline TileFusionManager.java:113]: Do not use unnecessary line.

File: TileCanvas.java [hunk 8]
```diff
@@ -0,0 +1,137 @@
...
+        graphics2D.fillRect(x, y, tileSize, tileSize);
+        graphics2D.setColor(Color.BLACK);
+        graphics2D.drawRect(x, y, tileSize, tileSize);
+        if (tileValue != 0) {
+            String text = String.valueOf(tileValue);
+            int textWidth = graphics2D.getFontMetrics().stringWidth(text);
+            int textHeight = graphics2D.getFontMetrics().getAscent();
+            graphics2D.drawString(text, x + (tileSize - textWidth) / 2, y + (tileSize + textHeight) / 2);
+        }
+    }
```
Comment (kritika-kumari-git) [inline TileCanvas.java:85]: Do not use documentation for the methods whose name describes the method itself.

**Review by kritika-kumari-git — CHANGES_REQUESTED** [review]

**Pull request closed without merging**